Extensions/Terraform/Src/Tasks/TerraformTaskV1/src/base-terraform-command-handler.ts

```diff
@@ -53,10 +53,10 @@ export abstract class BaseTerraformCommandHandler {
             cwd: tasks.getInput("workingDirectory")
         });
 
-        let countProviders = (commandOutput.stdout.match(/provider/g) || []).length;
+        let countProviders = ["aws", "azurerm", "google"].filter(provider => commandOutput.stdout.includes(provider)).length
         tasks.debug(countProviders.toString());
         if (countProviders > 1) {
-            tasks.warning("Multiple provider blocks specified in the .tf files in the current working drectory.");
+            tasks.warning("Multiple provider blocks specified in the .tf files in the current working directory.");
```

Comment:
This typo that you fixed also needs to be fixed in the unit tests for this function.

https://github.com/microsoft/azure-pipelines-extensions/search?q=Multiple+provider+blocks+specified+in+the+.tf+files+in+the+current+working+drectory&unscoped_q=Multiple+provider+blocks+specified+in+the+.tf+files+in+the+current+working+drectory

I would also add additional tests for the "inherited provider" scenario that I mentioned in my previous comment.

Reply:
I've updated the typo in the unit test as well now. Thanks for the heads up 👍

I'm not currently able to run the tests locally, so I'll wait to hear from the maintainers before adding new test cases.

Reply:
Yeah, it's not the end of the world, IMO. Thanks for submitting the PR! :)